### ISSUE TITLE
ProductVersion not set correctly in VS2015

### ISSUE THREAD

**Issue by distantcam** [dotnet/roslyn#5098]
Hi there.

My team has had a bug come up with different compiles of code between Visual Studio 2013 and Visual Studio 2015. In 2013 the ProductVersion of the file is set correctly if the `AssemblyInformationalVersion` is a non-standard version, but in 2015 the ProductVersion is 0.0.0.

Here's some code to reproduce the problem.

```
using System;
using System.Diagnostics;
using System.Reflection;

[assembly: AssemblyVersion("1.7.0.0")]
[assembly: AssemblyFileVersion("1.7.0.0")]
[assembly: AssemblyInformationalVersion("1.7.0-customstring")]

class Program
{
    static void Main(string[] args)
    {
        var versionInfo = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location);

        Console.WriteLine(versionInfo.FileVersion);
        Console.WriteLine(versionInfo.ProductVersion);
        Console.WriteLine("{0}.{1}.{2}", versionInfo.ProductMajorPart, versionInfo.ProductMinorPart, versionInfo.ProductBuildPart);
        Console.ReadLine();
    }
}
```
2013 output
![2013 output](https://cloud.githubusercontent.com/assets/1298847/9753802/55c8604a-56f4-11e5-89fd-dcdb77ab521f.png)

2015 output
![2015 output](https://cloud.githubusercontent.com/assets/1298847/9753825/7e476f5c-56f4-11e5-9cc2-c24912c3c1dc.png)

**Comment by enricosada** [comment]
I can repro, the old behaviour zeroed after the first invalid part of the product version ( non integer value ).

```csharp
[assembly: AssemblyVersion("1.0.0.0")]
[assembly: AssemblyFileVersion("1.0.0.0")]
[assembly: AssemblyInformationalVersion("1.2.a.4")]
```

build, and check version
```powershell
(Get-Item ".\ClassLibrary.dll").VersionInfo | fl *
```

result:
```
ProductBuildPart   : 0
ProductMajorPart   : 1
ProductMinorPart   : 2
ProductName        : ClassLibrary2
ProductPrivatePart : 0
ProductVersion     : 1.2.a.4
```

**Comment by enricosada** [comment]
If this is a bug (not a changed behaviour) i'll send a pr to fix it

**Comment by gafter** [comment]
@enricosada We would welcome your help with this! I do not believe the change was intentional.

The page for `AssemblyInformationalVersionAttribute` requires a warning from the compiler:

> Although you can specify any text, a warning message appears on compilation if the string is not in the format used by the assembly version number, or if it is in that format but contains wildcard characters. This warning is harmless.

I have not observed this warning on VS2013, so I don't think we need to add it.

**Comment by Giftednewt** [comment]
@gafter I've done some testing against VS2013 and VS 2015 and it appears as though this is also the case for [AssemblyFileVersionAttribute](https://msdn.microsoft.com/en-us/library/system.reflection.assemblyfileversionattribute(v=vs.110).aspx)

From what I can tell the VS2013 compiler is processing each character in the string and when a non integer value is found it stops processing and returns the version number with respect to the parts of the string that is has already processed. It appears that the Roslyn compiler bails as soon as it finds a non-numeric part in the version number string. See examples below for VS2013 output
 
![image](https://cloud.githubusercontent.com/assets/7267084/14427126/01818354-ffeb-11e5-93c2-2456fe0f57d1.png)

![image](https://cloud.githubusercontent.com/assets/7267084/14427493/fef62fe8-ffec-11e5-89e0-1256e6e705a1.png)

I also noticed that in VS2015 a warning is produced a warning for the format of the AssemblyFileVersionAttribute that is not present in VS2013.

![image](https://cloud.githubusercontent.com/assets/7267084/14427363/46b71730-ffec-11e5-9b07-2ba961487f4a.png)

Should this behaviour be re-introduced into the compiler and should the warning also be introduced for AssemblyInformationalVersionAttribute to be consistent with the documentation and the fact that it has been added for AssemblyFileVersionAttribute?

**Comment by Giftednewt** [comment]
@gafter,
I have an implementation of this which produces the same output as VS2013. My implementation also adds the warning mentioned for the AssemblyInformationalVersionAttribute. The changes I made are in this [commit](http://github.com/Giftednewt/roslyn/commit/9c1d4be3158853480c3570bbbe445e7f7ff8ce85)

I can PR it if you want?

**Comment by gafter** [comment]
Yes, please submit a PR and we'll review it. Please tag me on the PR.

**Comment by gafter** [comment]
@jaredpar Can you please confirm if you're OK with this being fixed in 1.3? That is the target of the current PR.

**Comment by gafter** [comment]
Fixed in #10620
